libstuff/libstuff.cpp

```diff
@@ -2805,25 +2804,83 @@ bool SIsValidSQLiteDateModifier(const string& modifier) {
     return true;
 }
 
-bool SREMatch(const string& regExp, const string& s) {
-    return pcrecpp::RE(regExp, pcrecpp::RE_Options().set_match_limit_recursion(1000)).FullMatch(s);
-}
+bool SREMatch(const string& regExp, const string& input, bool caseSensitive, bool partialMatch, vector<string>* matches) {
+    int errornumber = 0;
+    PCRE2_SIZE erroroffset = 0;
+    uint32_t matchFlags = 0;
+
+    // These require full-string matches as that's the historical way this function works.
+    uint32_t compileFlags = partialMatch ? 0 : PCRE2_ANCHORED | PCRE2_ENDANCHORED;
+    if (!caseSensitive) {
+        compileFlags |= PCRE2_CASELESS;
+    }
+    pcre2_match_context* matchContext = pcre2_match_context_create(0); 
+    pcre2_set_depth_limit(matchContext, 1000); 
+    pcre2_code* re = pcre2_compile((PCRE2_SPTR8)regExp.c_str(), PCRE2_ZERO_TERMINATED, compileFlags, &errornumber, &erroroffset, 0);
```

Comment:
Do we need to check for any failures from `pcre2_compile`? looks like it will return `null` if it failed

https://www.pcre.org/current/doc/html/pcre2demo.html
<img width="1031" alt="Screenshot 2024-07-26 at 11 20 11" src="https://github.com/user-attachments/assets/50ac00ec-36d4-43f1-bf84-83137117d03e">